raiden/transfer/architecture.py

```diff
@@ -0,0 +1,141 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=too-few-public-methods
+import types
+from collections import namedtuple
+from copy import deepcopy
+
+Iteration = namedtuple('Iteration', ('new_state', 'events'))
```

Comment:
> Are you attached to the Iteration nomenclature here?
Not at all.

`StateTransition` might be confusing because of the `StateChange` objects, any other suggestions?